bus-mapping/src/evm/opcodes/error_oog_dynamic_memory.rs

```diff
@@ -0,0 +1,38 @@
+use crate::{
+    circuit_input_builder::{CircuitInputStateRef, ExecStep},
+    error::{ExecError, OogError},
+    evm::Opcode,
+    Error,
+};
+use eth_types::{evm_types::OpcodeId, GethExecStep};
+
+#[derive(Clone, Copy, Debug)]
+pub(crate) struct OOGDynamicMemory {}
+
+impl Opcode for OOGDynamicMemory {
+    fn gen_associated_ops(
+        state: &mut CircuitInputStateRef,
+        geth_steps: &[GethExecStep],
+    ) -> Result<Vec<ExecStep>, Error> {
+        let geth_step = &geth_steps[0];
+        debug_assert!([OpcodeId::RETURN, OpcodeId::REVERT].contains(&geth_step.op));
+
+        let mut exec_step = state.new_step(geth_step)?;
+        exec_step.error = Some(ExecError::OutOfGas(OogError::DynamicMemoryExpansion));
+
+        state.stack_read(
```

Comment:
Scroll's PR differentiates between CREATE (three reads) and others (two reads). Any reason why there's no differentiation here?

Reply:
Oh, we moved CREATE oog case to a separted single gadget and we also did that in spec. Here is the PR for CREATE oog, https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1562#pullrequestreview-1579757261. I'll add this change in the description.